blaze/compute/tests/test_sparksql.py

```diff
@@ -357,11 +357,12 @@ def test_strlen(ctx, db):
 
 @pytest.mark.parametrize('attr', date_attrs)
 def test_by_with_date(ctx, db, attr):
+    # TODO: investigate CSV writing precision between pandas 0.16.0 and 0.16.1
     expr = by(getattr(db.dates.ds, attr),
               mean=db.dates.amount.mean())
-    result = odo(compute(expr, ctx), set)
-    expected = odo(compute(expr, {db: {'dates': date_df}}), set)
-    assert result == expected
+    result = odo(compute(expr, ctx), pd.DataFrame).sort('mean')
+    expected = compute(expr, {db: {'dates': date_df}}).sort('mean')
+    tm.assert_frame_equal(result, expected, check_dtype=False)
```

Comment:
hm it doesn't appear that `odo` will change the type of a `DataFrame`. I'll make an issue about it on `odo` and keep the test here as I have it